File: problems/algorithms/object-exam-oo/js/response/exam-check.js
class ExamCheck {
  constructor(answer, weight) {
    this.weight = weight;
    this.answer = answer;
  }

  grade(student) {
    return Object.keys(student).reduce((total, q) => {
      return total + (student[q] === this.answer[q] ? this.weight[q] : 0);
    }, 0);
  }
}

export { ExamCheck };

File: problems/algorithms/object-exam-oo/ruby/response/src/exam_check.rb
class ExamCheck
  def initialize(answer, weight)
    @weight = weight;
    @answer = answer;
  end
  def grade(student)
    asserts = 0
    student.each do |key, value|
      asserts = student[key] == @answer[key] ? asserts + @weight[key] : asserts + 0
  end
  return asserts
  end
end
